### PR TITLE
Throws error if invalid multi-word argument passed into packager

### PR DESCRIPTION
**Pull request by ronalag**
The issue related to this bug fix is:

blackberry/BB10-Webworks-Packager#138

I made the packager throw an error if a user passes an invalid multi-word command line argument

@EricLeiLi could you please do a code review?

### CODE REVIEW

File: lib/cmdline.js [hunk 1]
```diff
@@ -28,16 +29,35 @@ cmd
...
+        if (args[i].indexOf("--") === 0 && !command[option]) {
+            throw localize.translate("EXCEPTION_CMDLINE_ARG_INVALID", args[i]);
+        }
+    }
+
+    return this;
+}
+
+module.exports = {
+    obj: command,
```
Comment (jamesjhedley) [inline lib/cmdline.js]: @ronalag Really like the changes you made. I wonder if we should get rid of the obj method and simply make parse return the parsed object.

File: lib/cmdline.js [hunk 2]
```diff
@@ -28,16 +29,35 @@ cmd
...
+        }
     }
-}
     
-module.exports = cmd;
+    command.parse(args);
+
+    //Check for any invalid command line args
+    for (i = 0; i < args.length; i++) {
+        option = args[i].substring(2);
```
Comment (jamesjhedley) [inline lib/cmdline.js]: Might add a quick comment here //Remove leading dashes [--]

File: test/unit/lib/cmdline.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
 var srcPath = __dirname + "/../../../lib/",
+    localize = require(srcPath + "localize"),
+    cmdline = require(srcPath + "cmdline"),
```
Comment (jamesjhedley) [inline test/unit/lib/cmdline.js:3]: The logic here is a little strange. Having both a "cmdline" and "cmd" and both being the command-line module.

**Comment by nukulb** [comment]
@issp - are you suppose to test this?

**Comment by issp** [comment]
Igor is testing this issue.

**Comment by ishneur** [comment]
All CI builds for this branch failed: http://mac-ci:9000/job/BB10-Webworks-Packager-next-fix138-cmdopt/

**Comment by ronalag** [comment]
@ishneur  The CI builds for this issue are working

**Comment by ishneur** [comment]
Verification passed on Windows and Mac using CI build: http://mac-ci:9000/job/BB10-Webworks-Packager-next-fix138-cmdopt/5/. The error message now is: [ERROR]   Invalid command line argument "--src".

Thanks to @tracyli for verifying on the Mac.

**Comment by nukulb** [comment]
@ronalag - its fine for now but unrelated commits should come in as different pull requests.

**Comment by nukulb** [comment]
@jkeshavarzi @ronalag - what is left with the code review here?

**Comment by jamesjhedley** [comment]
The comments I put were the only issues I saw. Henry and I were discussing them yesterday.

Everything else looked good.

----- Original Message -----

**Comment by nukulb** [comment]
@ronalag - please address @jkeshavarzi 's comments so that we can close this

**Pull request merged**